src/test/java/com/cag/cagbackendapi/util/SpringCommandLineProfileResolver.kt

```diff
@@ -0,0 +1,17 @@
+package com.cag.cagbackendapi.util
+
+import org.springframework.test.context.ActiveProfilesResolver
+import org.springframework.test.context.support.DefaultActiveProfilesResolver
+
+class SpringCommandLineProfileResolver : ActiveProfilesResolver {
+    private val defaultActiveProfilesResolver = DefaultActiveProfilesResolver()
+
+    override fun resolve(testClass: Class<*>): Array<String> {
+        val springProfileKey = "spring.profiles.active"
+        return if (System.getProperties().containsKey(springProfileKey) && System.getProperty(springProfileKey).isNotEmpty())
+            System.getProperty(springProfileKey).split("\\s*,\\s*".toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray()
+        else
+            defaultActiveProfilesResolver.resolve(testClass)
+    }
+}
+
```

Comment:
I am unsure on what this file does.

Reply:
When we're using test classes, this SpringCommandLineProfileResolver helps us set the correct Spring profile to set the environment variables in the testing class correctly. All the environment variables and spring profiles are stored in the application.yml file!